### DISCUSSION TITLE
Ability to disable cache:clear_php_http

### DISCUSSION THREAD

**Discussion by mikestreety** [sourcebroker/deployer-extended#17] [Q&A] [answered]
Hello!

I use your [deployer-extended-typo3](https://github.com/sourcebroker/deployer-extended-typo3) package which, in turn, uses this calls `cache:clear_php_http`.

In some instances I don't want this to be called - if the site is behind a http auth or if it doesn't have a live URL yet, however the only way I can disable it is by overwriting the whole function with 

```php
task('cache:clear_php_http', function () {
	
});
```

In the times I don't want it, I don't want to have to redeclare the whole `deploy` task to just remove it, so t would be really nice if there could be an option to disable this function or it not throw an error if `public_urls` is missing.

Thanks, as always, for all the work!

**Comment by kszymukowicz** [comment]
I was thinking the same in past. But for setting the disable flag you would anyway need to use a one line code like: `set('enable_cache_clear_php_http', false)`;

For deployer 7.* you can however just do:
`task('cache:clear_php_http')->disable();`

I am not sure there can be anything simpler. A flag for that would be making things more complex where there is no need to.
What do you think?

**Comment by mikestreety** [comment]
My use-case is that I have a central package which includes yours and sets a few defaults which we use on all of our projects. 

As some of our projects are behind http auth, I disabled the `cache:clear_php_http` function (using the original method which was the "done thing" in Deployer 6.

My issue arose when I wanted the task to fire for one of my projects, so a flag would be preferable (especially if it could be set per environment).

For now, I have redefined the `deploy` task to exclude this task and then add the following in my local `deploy.php`

```php
on(select('instance=production'), function ($host) {
	after('cache:clear_php_cli', 'cache:clear_php_http');
});
```

The downside now is that if you add a new task, I don't benefit from it.

I appreciate this is an edge case and is obviously not a common problem as no-one else has reported it :)

My preference would be for it to run if `public_urls` is set as it seems to be the only task that uses it.

**Comment by kszymukowicz** [comment, marked as answer]
What about:

```task('cache:clear_php_http')->select('cache_clear_php_http_task_disable!=true');```

and then where you define host

```
   host('staging')
    ->add('labels', ['cache_clear_php_http_task_disable' => 'true'])
```

**Comment by mikestreety** [comment]
@kszymukowicz Yeah, that works :)

**Comment by einNicklas** [comment]
Hey @kszymukowicz,

what do you think about the possibility to make it possible to configure basic auth credentials?

I have the same problem: a staging system, so the cache:clear_php_http fails.

It would be possible to pass the credentials to wget:

```sh
wget --user user --password pass http://example.com/
```

Best regards
Nicklas

**Comment by kszymukowicz** [comment]
@einNicklas 

Sure! 
if you will prepare a PR for all 3 options to make a request (I mean not only wget) then I will merge.

**Comment by einNicklas** [comment]
I just had a look at the code. There is already a possibility for wget and curl:

```php
host('staging')
    ->set('fetch_method_wget_options', '--user user --password password')
```

For `file_get_contents` you would then need a http header stream context.

Unfortunately, I don't have the time to get to grips with this at the moment.
